rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/Versioned.java

```diff
@@ -31,10 +31,12 @@
 public class Versioned<T> {
     private final int version;
     private final T object;
+    private final long timestamp;
 
     public Versioned(int version, T object) {
         this.version = version;
         this.object = object;
+        this.timestamp = System.currentTimeMillis();
```

Comment:
we have no guarantee that the versions are monotomic in their construction. so version 10 might be constructed on thread A at time 20, while version 9 got scheduled on thread B at time 22.

This gets especially messy when `tpl` files are around where we depend on the "newer" relationship. What I'm wondering is if we could, register the timestamp as close to the time we receive the `didChange` message? (it might be close to here, I haven't checked)